napari/layers/_labels_layer/model.py

```diff
@@ -507,6 +501,19 @@ def get_message(self, coord, value):
 
         return msg
 
+    def _update_thumbnail(self):
+        """Update thumbnail with current image data and colors.
+        """
+        zoom_factor = np.divide(self._thumbnail_shape[:2],
+                                self._image_view.shape[:2]).min()
+        zoomed = np.round(ndi.zoom(self._image_view, zoom_factor,
+                                   prefilter=False, order=0))
+        zoomed = self.raw_to_displayed(zoomed)
+        mapped = self.colormap.map(zoomed) * 255
+        mapped = mapped.reshape(list(zoomed.shape) + [4])
+        mapped[:, :, 3] = mapped[:, :, 3] * self.opacity
```

Comment:
is this different if you did
```python
mapped[:, :, 3] *= self.opacity
```
?

Reply:
even though we know this is a 2D image, I think it's nicer to use the nD idiom:

```suggestion
        mapped[..., 3] *= self.opacity
```

Reply:
fixed